lib/driver.js

```diff
@@ -233,6 +238,42 @@ class EspressoDriver extends BaseDriver {
     }
   }
 
+  /**
+   * Unzip the given app path if needed.
+   *
+   * @param {string} appPath The path to app file.
+   * @returns {?string} Retuns the path to an unzipped app file path
+   *                    if the given path was unzipped, otherwise null.
+   */
+  async unzipAppIfNeeded (appPath) {
+    if (path.extname(_.toLower(appPath)) !== ZIP_EXT) {
+      return null;
+    }
+
+    const useSystemUnzipEnv = process.env.APPIUM_PREFER_SYSTEM_UNZIP;
+    const useSystemUnzip = _.isEmpty(useSystemUnzipEnv)
+      || !['0', 'false'].includes(_.toLower(useSystemUnzipEnv));
+    const tmpRoot = await tempDir.openDir();
+    await zip.extractAllTo(appPath, tmpRoot, {useSystemUnzip});
+
+    const globPattern = `**/*.+(${SUPPORTED_EXTENSIONS.map((ext) => ext.replace(/^\./, '')).join('|')})`;
+    const sortedBundleItems = (await fs.glob(globPattern, {
+      cwd: tmpRoot,
+      strict: false,
+    })).sort((a, b) => a.split(path.sep).length - b.split(path.sep).length);
+    const unzippedAppPath = path.join(tmpRoot, _.first(sortedBundleItems));
```

Comment:
there is still no verification for the case `sortedBundleItems` is empty

Reply:
added when the zip does not have valid extension packages case